### ISSUE TITLE
windows 10 reports trojan in process hacker

### ISSUE THREAD

**Issue by martinlindhe** [winsiderss/systeminformer#454]
Just checked for update on process hacker and windows told me this:

![ph-trojan](https://user-images.githubusercontent.com/181531/62832032-ca48e780-bc28-11e9-98ad-7ccdd3804f2b.png)

**Comment by ExE-Boss** [comment]
For me, it says: `Trojan:Win32/Occamy.C`

**Comment by Biswa96** [comment]
Just ignore. There are 8-9 similar issues.

**Comment by ExE-Boss** [comment]
I’ve submitted the file to Microsoft using: https://www.microsoft.com/en-us/wdsi/filesubmission.

Hopefully, it will get resolved soon.

**Comment by martinlindhe** [comment]
So can someone confirm this is a false positive?

**Comment by ExE-Boss** [comment]
Microsoft have now confirmed that it indeed isn’t a virus, and it will be whitelisted in the next **Windows Defender** update:
![image](https://user-images.githubusercontent.com/3889017/62868946-15412880-bd17-11e9-9ca4-4107c25e1210.png)

**Comment by dmex** [comment]
5th of Aug - Sehyioa.A!cl
12th of Aug - Occamy.C
12th of Aug - Wacatac.B!ml


![image](https://user-images.githubusercontent.com/1306177/62875225-fd44c580-bd65-11e9-8604-4c6feb113dbf.png)

![image](https://user-images.githubusercontent.com/1306177/62875236-033aa680-bd66-11e9-8b8e-14ae13ad4e0a.png)

![image](https://user-images.githubusercontent.com/1306177/62875244-07ff5a80-bd66-11e9-9acc-d81790f21773.png)

**Comment by Biswa96** [comment]
Is there any information what those virus do or it's attacking methods? So one can check if those signatures match with ProcessHacker.

**Comment by Bartolomeus-649** [comment]
Perhaps one could submit checksums to Virus Total, they seem to have a program to eliminate false possitives from AV-Tools.
https://blog.virustotal.com/2015/02/a-first-shot-at-false-positives.html

Or at least, submit any new executables released to their  online service which uses 67 anti virus engines to check for malware.

One could (probably) use an url like this to submit a file:
```
https://www.virustotal.com/ui/search?relationships%5Bcomment%5D=author%2Citem&relationships%5Burl%5D=network_location%2Clast_serving_ip_address&limit=20&query=<UrlEncodedUrlToFileToSubmit>
```
this results in a result page looking something like this:
https://www.virustotal.com/gui/url/988e8bbea82cc2caa92267ba560c8b25cfacebdb2fe642e087027329c2e4c324/detection

Virus Total also has an API: `https://developers.virustotal.com/reference` which is probably the prefered way of interacting with them.

**Comment by ralvarador** [comment]
Well, Windows 10 AV (defender? ) again claiming ProcessHacker as a Virus!, as a TROJAN VIRUS; this time on of its DLLs.
![imagen](https://user-images.githubusercontent.com/30580377/68095499-1302ec00-fe78-11e9-8fde-8e474932c7f3.png)

**Comment by dmex** [comment]
Closing as external 👍

**Comment by letrec** [comment]
Trojan:Win32/Casdet!rfn:
![PH](https://user-images.githubusercontent.com/996569/68681343-3592b480-059e-11ea-8f17-9df8634325f6.png)

**Comment by letrec** [comment]
Microsoft confirmed it to be false positive:
![PH2](https://user-images.githubusercontent.com/996569/68756988-22d3ba80-0646-11ea-8a98-34e7ad3309a8.png)
This is for Process Hacker vrsion 3.0.2780.

**Comment by Trojaner** [comment]
![grafik](https://user-images.githubusercontent.com/1809172/69542420-76c29580-0f9c-11ea-9906-d81fae00e3c3.png)

![grafik](https://user-images.githubusercontent.com/1809172/69542433-7e823a00-0f9c-11ea-99e0-d4d31f6a3f83.png)

**Comment by dmex** [comment]
I've asked Microsoft asking for more information.

**Comment by ExE-Boss** [comment]
Above error when translated roughly means:

<pre>
HackTool: Win64/ProcHack
Warning Level: High
Status: Active
Date: 25.11.2019 15:56
Category: Tool
Details: The behaviour of this program is potentially undesirable.

[Related Information]

Affected Items:
- file: E:\Program Files\Process Hacker 2\kprocesshacker.sys
</pre>

This&nbsp;is&nbsp;probably to&nbsp;be&nbsp;expected, since&nbsp;`kprocesshacker.sys` is&nbsp;a&nbsp;kernel driver that allows **Process&nbsp;Hacker** to&nbsp;provide some extra features that&nbsp;require kernel&nbsp;access, which&nbsp;**Windows** places extra protection&nbsp;on.

---

Does&nbsp;`kprocesshacker.sys` allow&nbsp;**Process&nbsp;Hacker** to&nbsp;do&nbsp;stuff that&nbsp;normally requires&nbsp;elevation without&nbsp;being itself&nbsp;elevated?

Because&nbsp;if&nbsp;so, then&nbsp;that&nbsp;might be&nbsp;why: “the&nbsp;behaviour of&nbsp;this&nbsp;program is&nbsp;potentially&nbsp;undesirable”.

**Comment by dmex** [comment]
> Does kprocesshacker.sys allow Process Hacker to do stuff that normally requires elevation without being itself elevated?

You can't load the driver without (elevated) administrative access and the driver does not allow connections or requests from non-elevated processes. An administrator can deliberately disable those protections (patching files/registry keys etc...) and make the current Windows configuration insecure but every one of those cases requires administrative access.

**Comment by cliffordx** [comment]
![2019-11-28_19-38-51](https://user-images.githubusercontent.com/7141739/69803473-37ec4400-1217-11ea-9bc2-88bab05122ec.jpg)
Whitelisting did the trick

**Comment by TNTUP** [comment]
Mine says TrojanDownloader:O97M/Emotet.RO!MTB (Windows Server 2016) not sure if its really safe, been using ProcessHacker since 2-3 years... I'm a paranoid guy so I could whitelist it but it is a false positive I think

**Comment by Trojaner** [comment]
@TNTUP 
Emotet is a very dangerous virus, maybe it is indeed a real Trojan hiding as Process Hacker. I suggest downloading process hacker again, from an official download link.

**Comment by TNTUP** [comment]
> @TNTUP
> Emotet is a very dangerous virus, maybe it is indeed a real Trojan hiding as Process Hacker. I suggest downloading process hacker again, from an official download link.

I downloaded it off processhacker.sourceforge.io, probably thats shady but thats the same file I have on my storage server o_o

**Comment by hizamakura** [comment]
> > @TNTUP
> > Emotet is a very dangerous virus, maybe it is indeed a real Trojan hiding as Process Hacker. I suggest downloading process hacker again, from an official download link.
> 
> I downloaded it off processhacker.sourceforge.io, probably thats shady but thats the same file I have on my storage server o_o

Check the hash of the .exe file on virustotal, it should be the one with +300 community score.
And also, I got this error in Windows 10 Version 2004 Build 19033.1

Same error as @cliffordx ![Image](https://chito.ge/4oSaQmk.png)

**Comment by ygoe** [comment]
This certainly isn't an issue of manipulated files. I've downloaded the setup package on 2016-06-04 and reinstalled it after Windows Defender ate it. Then it ate it again right away. I hope my exclusion works now. Otherwise I'll have to consider disabling Windows Defender completely. Antivirus software never found something here, just deleted or blocked needed files. I don't need that.

**Comment by ddbb2017** [comment]
It's not resolved yet. Someone at Microsoft needs to get their $h*t together, I could run Process Hacker just fine, but when I navigated to its installation folder in Windows Explorer it showed this:

![Capture0](https://user-images.githubusercontent.com/33359889/70474668-e4d08780-1a87-11ea-8707-e950675cd6fd.PNG)

and deleted the .sys file. The version of the driver is 3.0.0.0 signed by `Wen Jia Liu` on `‎Monday, ‎March ‎28, ‎2016 10:21:05 AM`. The serial # for the cert is `040cb41e4fb370c45c4344765162582f`

Then on my other computer, I was getting this message for `ProcessHacker.exe` v.2.39.0.124:

![Capture1](https://user-images.githubusercontent.com/33359889/70474961-67594700-1a88-11ea-82ca-8009d5072eb5.PNG)

which was also deleted by Windows Defender, for being a `"tool"`? :)

**Comment by zmni** [comment]
Also same issue on Windows 8.1 x64:

![processhacker](https://user-images.githubusercontent.com/841523/70526024-39086400-1b7b-11ea-92c5-c9961bcf5d7b.jpg)

**Comment by krokofant** [comment]
Cool stuff. Suddenly Process Hacker was missing because Windows Defender removed the exe 😆I didn't even notice it.

**Comment by hl2guide** [comment]
For me as of today it is showing as:

![a](https://user-images.githubusercontent.com/17044982/70659186-aa4b3280-1cb3-11ea-8c0d-b36ad6512f77.PNG)

**Comment by ygoe** [comment]
After I have configured Process Hacker as exception in Defender Antivirus, it was now deleted for the second time. I consequently have to **consider Defender AV as malware** and deactivated it entirely. Never did anything for me anyway. I have reported this twice as a severe problem of an important feature through Windows Feedback and would advise anybody to do the same.

**Comment by speedwaystar** [comment]
Windows Defender now seems to be generating a new quarantine for Process Hacker every other day, despite being told the previous day to allow it. It's possible that this (re)occurs whenever it downloads a new threat list from WIndows Update.

![](https://i.imgur.com/XePOon9.png)

It's doubly annoying because when Windows Defender is told to "allow" Process Hacker, it says "This threat has been allowed and will not be remediated in the future." Sadly this is an outright lie, since it has now remediated it for three days running.

**Comment by ygoe** [comment]
Update: Disabling Windows Defender is not effective. After restarting Windows, it is automatically enabled again – and does its thing. Also, stopping, disabling or deleting the service is not possible. The action itself is prohibited for Administrators (when done with Process Hacker) and Windows Updates are likely to restore it anyway. A work-around that I found is adding all of your hard drives as exclusions. Add C:\, D:\ and whatever drives you have as excluded folders. I'll report if this also doesn't help to keep Process Hacker on the computer.

And, please keep reporting this as negative feedback (severe problem) in the Defender app. Microsoft has to read the feedback eventually.

**Comment by Trojaner** [comment]
@ygoe you can disable Windows Defender permanently by setting DisableAntiSpyware to 1 in your registry (I don't know the exact path, Google it)

**Comment by ExE-Boss** [comment]
I’d just whitelist the entire **ProcessHacker** directory.

**Comment by VimalShekar** [comment]
In this particular case, Defender is blocking you for a valid reason. 
(I'm not trying to be mean here, there really is a reason). 

The owners of this project have to submit an issue to Microsoft using 
https://www.microsoft.com/en-us/wdsi/filesubmission

MS would analyse and respond back. If they say the detection is intended, you should then raise a "dispute". At this point, they would reveal their reasons and help you fix the code that's causing them to flag it. 

If you have just submitted an issue and haven't gone through the "dispute" process, you will never know the reason why it was flagged.

**Comment by dylmye** [comment]
@VimalShekar this has been done before :( https://github.com/processhacker/processhacker/issues/454#issuecomment-520428285

**Comment by dmex** [comment]
@VimalShekar 

> The owners of this project have to submit an issue to Microsoft

I've done this about 8 times this year.

> If you have just submitted an issue and haven't gone through the "dispute" process, you will never know the reason why it was flagged.

My account was banned by Microsoft the same day they pushed the detection and I can't dispute it.

There was an email conversation with Microsoft about the issue but they haven't replied about that or provided information about anything.

**Comment by ddbb2017** [comment]
@dmex yeah, sounds like a typical MSFT. (Basically when the right hand doesn't know what the left one is doing.)

Try to create another Microsoft account and see if you can find out why they banned it. My guess is that Process Hacker interferes with the functionality of one of their products, i.e. Process Explorer. In that case they won't let you know. But if it's just a false-positive, say, for calling too many native functions, then it's another matter.

**Comment by ddbb2017** [comment]
Maybe [tweet at them](https://twitter.com/ahmdweb/status/1206690127457599488) too.

**Comment by ygoe** [comment]
@ddbb2017 Retweeted. Our little community is probably too small to create a shitstorm.

**Comment by VimalShekar** [comment]
> @VimalShekar this has been done before :( [#454 (comment)](https://github.com/processhacker/processhacker/issues/454#issuecomment-520428285)

That's August 11, for a different issue... 
The detection you are seeing now was added in November for a recent issue...

**Comment by dylmye** [comment]
Reply from John Lambert from Microsoft Threat Intelligence Center:

> We are aware of this and last I heard the Defender team was going to reach out to Process Hacker folks. If you hear this hasn't happened, please have someone ping us! 

[(source)](https://twitter.com/JohnLaTwC/status/1206714605642190848)

**Comment by dmex** [comment]
@dylmye 

I can confirm Microsoft has contacted the development team.

**Comment by luckydonald** [comment]
Now I'm curious why they deleted your account.

**Comment by ddbb2017** [comment]
@dmex ok, so what's the rub? What did they not like in there?

**Comment by dmex** [comment]
@ddbb2017 

It's an active conversation and would not be appropriate to share that information right now. I'll post an update after the discussion with Microsoft has concluded.

**Comment by ddbb2017** [comment]
@dmex ok, fair enough, please do.

**Comment by dmex** [comment]
"conversation" is the wrong word to describe whats happening at Microsoft right now.

**Comment by dylmye** [comment]
We can loop in John and Tanmay from MDATP if that helps at all.

**Comment by ddbb2017** [comment]
@dylmye before anything can be done, can someone please explain what do they want from this project? @dmex I'm lost af.

**Comment by dmex** [comment]
@dylmye 

If you think it would help, feel free to ask them.

@ddbb2017 

A riddle wrapped in an enigma... Microsoft hasn't provided any actual details, only some vague claims and a request we change something that was changed a long time ago.

> Now I'm curious why they deleted your account.

My account is still banned.

**Comment by ddbb2017** [comment]
I bet they want you to re-sign the driver. and pass it thru their WHQL.

**Comment by poizan42** [comment]
Are they unhappy about it being able to inspect and manipulate protected processes? But then they should flag kprocesshacker.sys and not processhacker.exe.

**Comment by gchq-zz** [comment]
As of right now it is being passed by the MS engine at VirusTotal - took some thumping on the door, but it looks like someone finally opened it and listened :-)

**Comment by TNTUP** [comment]
Windows Defender does not flag Process Hacker anymore. Was able to download 2.39 and install and running as normal. Works on WS 2016 :)

**Comment by dmex** [comment]
Microsoft declared the  PE Viewer utility included with Process Hacker as a Trojan and silently removed the binary and start-menu shortcuts from user' machines.

The only way to restore the peview utility is to completely reinstall. #544

-----------

The telemetry reports at minimum **1.7 million** installations of Process Hacker have been corrupted and/or removed by Microsoft to date... Netscape 2.0 - bill gates would be impressed.

![image](https://user-images.githubusercontent.com/1306177/73121636-e6b6c180-3fc7-11ea-948a-dbab2b2178af.png)

**Comment by ddbb2017** [comment]
@dmex what? they're at it again.

As a developer don't you "love" those AV companies?

**Comment by luckydonald** [comment]
This is very lovely.

**Comment by letrec** [comment]
@dmex, Actually there's a way to restore it without reinstalling the application. One needs to go to Windows Security -> Virus & treat protection -> Protection history, identify the right "Threat blocked item" and click the button to unblock it.
![Virus   treat protection](https://user-images.githubusercontent.com/996569/73129133-b139c480-4017-11ea-955d-d135462e3374.png)
![Protection history](https://user-images.githubusercontent.com/996569/73129151-fb22aa80-4017-11ea-9a51-b63380b24ad4.png)

**Comment by dmex** [comment]
Microsoft has released 3 additional detections for Process Hacker:

"We have determined that the files meet our criteria for detection. At this time detection will remain in place. "

- OnlineChecks.dll - Trojan/Win32/Detplock
- ProcessHacker.exe - Trojan/Win32/Casdet!rfn
- ProcessHacker-setup.exe - Trojan/Win32/Wacatac.C!ml

https://www.microsoft.com/en-us/wdsi/submission/645d8857-7058-4e89-b329-8b358f1fdccc

![image](https://user-images.githubusercontent.com/1306177/75208743-46bba600-57c8-11ea-8c91-bb524de4c1fe.png)

**Comment by namazso** [comment]
I also had the Detplock / Wacatac detections on my software, seems like Microsoft just detects absolutely everything that is unsigned now

**Comment by RossWang** [comment]
Isn't the original author work at Microsoft?
Oh no, he left one year ago!

![image](https://user-images.githubusercontent.com/13231557/75310253-62897f80-588e-11ea-92af-9e1c953c0145.png)

As I said in issue #408, hackers login to RDP and use PH to turn off security software. What interested me is a recent release of **Kaspersky Endpoint Security** blocks PH in a clever way—**only detects PH in RDP sessions**.

**Comment by dmex** [comment]
> Isn't the original author work at Microsoft?

Yes. The project author and founder was a senior engineer for the Microsoft Hyper-V team.

> Oh no, he left one year ago!

He has left yes. I don't know the details but it's unfortunate and I hope it wasn't because other employees at Microsoft have been attacking his projects and labeling everything as "trojans" or sending him threats,,. It's probably not appropriate to speculate since its unrelated to this issue and I'm biased as to the reasons after employees at Microsoft emailed me a few years ago with threats and more recent actions by Microsoft towards the project.

> As I said in issue #408, hackers login to RDP and use PH to turn off security software.

This is not what Microsoft or anyone else has been claiming with their descriptions and signatures?

You also can't disable security software using Process Hacker... Terminating a process does not stop the process from restarting and does not uninstall the product and also does not even disable kernel drivers, remove kernel object callbacks or remove disk minifilter callbacks - all of these and other security related callbacks, functionality and policies cannot be disabled with Process Hacker.

You're probably confusing Process Hacker with how attackers are actually exploiting RDP flaws and the access level provided by RDP (which isn't available for home users) and how it bypasses literally every security policy and mitigation on Windows - I can't fix the security issues with RDP because its a Microsoft product and can only recommend you don't ever use it over the internet (which it does not provide any protection and was never designed for).

For example; If you execute this powershell script over RDP and it'll permanently destroy all security software (and the entire operating system) with a single built-in Windows command: https://gist.github.com/dmex/5885ec528b8231fb75d5a591845b96e6

**Comment by dmex** [comment]
Microsoft has released 5 new signatures for Process Hacker today: 

    Trojan:Win32/Wacatac.D!ml
    Trojan:Win32/Wacatac.C!ml
    Trojan:Win32/Wacatac.B!ml
    Trojan:Win32/Fuery.C!cl
    Trojan:Win32/Detplock

These are additional to the 3 signatures from yesterday:

    OnlineChecks.dll - Trojan/Win32/Detplock
    ProcessHacker.exe - Trojan/Win32/Casdet!rfn
    ProcessHacker-setup.exe - Trojan/Win32/Wacatac.C!ml

**Comment by dylmye** [comment]
Contacted MDATP again, no luck (yet):

https://twitter.com/dylmye/status/1232594687090941953

**Comment by ddbb2017** [comment]
@dmex wtf, dude? why tf would you post a scrambled PowerShell script here to delete someone's boot partition? It really doesn't help your case. 

@dylmye  Did anyone actually look into the source code that MSFT are flagging?

**Comment by dmex** [comment]
@ddbb2017 

> why tf would you post a scrambled PowerShell script here to delete someone's boot partition? 
> It really doesn't help your case

The truth dies in darkness when you dismiss legitimate discussion like that.  That's not my code. I found it while investigating an attack and it should be discussed instead of hidden away.

A simple script that bypasses all security software and causes considerable damage is I would think highly relevant to our case since people have been accusing Process Hacker of RDP attacks and damage oblivious to the RDP vulnerabilities causing the security issues and resulting damage in the first place.

Process Hacker is not some magical unicorn that is able to bypass security privileges and security policies or exploit Windows - It calls the exact same functions used by Microsofts own products and can only do something when you have privileges to preform that action just like Microsofts products.

If you do something you think is 'bad' with Process Hacker it's because you have the privileges and the right defined by group policy to use those functions - if you don't want Process Hacker doing something then change your group policies and account privileges it's that simple - something else can come along and do exactly what Process Hacker does unless you configure your polices - singling out and targeting Process Hacker won't help anyone and just demonstrates incompetence and anti-competitive behavior.

Microsoft is the only one that can stop RDP attacks by improving the security and authentication support of RDP... There is nothing else that we can do that I know about. I've already added every last security mitigation and security feature available on Windows into Process Hacker - I literally can't add anything more unless Microsoft changes their policies allowing open-source projects to use PPL and CI (which they already use with the Windows store for Games and Calculator apps)... 

Maybe one day when Microsoft stops attacking the project and changes their policies...

![image](https://user-images.githubusercontent.com/1306177/75435931-7cff4e00-599f-11ea-97e8-bacecf185c8d.png)

> hackers login to RDP and use PH to turn off security software

In addition to my last comment about that...

What happens when you terminate dwm.exe? It restarts.
What happens when you terminate winlogon.exe? It restarts.
What happens when you terminate explorer.exe? It'll restart when you use status 0.
What happens when you terminate lsass.exe? It shows a message for 60s before rebooting (you can cancel the reboot using shutdown.exe /a)
What happens when you terminate csrss.exe? It fails a handle check and ntoskrnl.exe bug-checks the machine with an error code.

What happens when you terminate security processes? Nothing. Just blame Process Hacker.

Why doesn't security software shutdown the machine or restart the process like everything else? Lazy idiots and a convenient excuse to target Process Hacker which they see as a competitor after we integrated 71 antivirus products and the ability to terminate threats without requiring constant daily updates and monthly subscriptions.

Security software is implemented in the kernel and even after terminating the process it'll continue running. You can check using Windows-Kernel-Explorer and viewing the various callbacks:
https://github.com/AxtMueller/Windows-Kernel-Explorer

Microsoft and other companies are deliberately choosing the worst possible way at handing this issue and the only thing I can summarize is they would rather destroy the project than do anything constructive and let open-source projects to use PPL and CI.

**Comment by namazso** [comment]
![image](https://user-images.githubusercontent.com/8676443/75438099-abf1d180-5957-11ea-90b8-697f0a1b8ddf.png)
meanwhile it seems like detections were cleared once again

**Comment by dylmye** [comment]
Can confirm @namazso - https://www.microsoft.com/en-us/wdsi/submission/645d8857-7058-4e89-b329-8b358f1fdccc

**Comment by ygoe** [comment]
May I ask what is "PPL and CI"? Google doesn't know it.

**Comment by ddbb2017** [comment]
@ygoe **PPL** = Protected Process Light, **CI** = Code Integrity.

At this stage, [both are available](https://docs.microsoft.com/en-us/windows/win32/services/protecting-anti-malware-services-) for Microsoft products only (or, through a kernel hack.) CI is enabled by default for Windows Store apps, but as a developer you won't be able to do much with it afterwards.

@dmex my point was not about worthiness of PH. (I agree with you on that.) What I was trying to tell you is that linking to an obfuscated malware sample that can wipe out someone's boot drive (if they run your power shell script) is not the best of options to prove what we're trying to prove here. That's it.

As a developer myself I totally hear you. I hate most AVPs and their false detections. But that's the world we have to live in since consumers seem to rely on them.

**Comment by Bartolomeus-649** [comment]
Interested reading that might be of some help...

- **Rick Strahl: [Dealing with Anti-Virus False Positives](https://weblog.west-wind.com/posts/2016/oct/05/dealing-with-antivirus-false-positives)**
https://weblog.west-wind.com/posts/2016/oct/05/dealing-with-antivirus-false-positives

- **Microsoft docs: [How Microsoft identifies malware and potentially unwanted applications](https://docs.microsoft.com/en-us/windows/security/threat-protection/intelligence/criteria)**
https://docs.microsoft.com/en-us/windows/security/threat-protection/intelligence/criteria

- **Microsoft Defender ATP Research Team: [Partnering with the industry to minimize false positives](https://www.microsoft.com/security/blog/2018/08/16/partnering-with-the-industry-to-minimize-false-positives/)**
https://www.microsoft.com/security/blog/2018/08/16/partnering-with-the-industry-to-minimize-false-positives/

- **Windows -> Security -> Threat protection:  [Information for developers](https://docs.microsoft.com/en-us/windows/security/threat-protection/intelligence/developer-info)**
https://docs.microsoft.com/en-us/windows/security/threat-protection/intelligence/developer-info

**Comment by Bartolomeus-649** [comment]
Also, it seem like the certificate used has expired...
![image](https://user-images.githubusercontent.com/7513795/75493691-c4df9e80-59ba-11ea-980d-45d164edc366.png)

**Comment by gchq-zz** [comment]
@Bartolomeus-649 

What age is the file? It's not like an SSL Cert, the valid date is only relevant during publishing! According to your screenshot it was signed on 09 April 2016, but was valid at the time until Jan 2017
![PHacker_002](https://user-images.githubusercontent.com/3681474/75495376-38f65400-5973-11ea-85ec-405e2026047b.png)


![PHacker_003](https://user-images.githubusercontent.com/3681474/75495383-401d6200-5973-11ea-9144-7170fae95e30.png)


![PHacker_004](https://user-images.githubusercontent.com/3681474/75495390-44497f80-5973-11ea-835f-3ab136b0b683.png)

**Comment by ddbb2017** [comment]
@Bartolomeus-649 that's why that signature has a timestamp that makes it valid, even today, when the certificate itself has expired. That's the whole point of timestamps.

As for re-signing it, then they can't. For that they will need a special certificate to sign a 64-bit kernel driver. That is done with an EV code signing certificate. I'm not sure about the further steps. It may also require [passing it through the WHQL](https://docs.microsoft.com/en-us/windows-hardware/drivers/dashboard/register-for-the-hardware-program).

**Comment by namazso** [comment]
@Bartolomeus-649  as you can see it is countersigned by the DigiCert time server, with a date of 2016. As far as I'm aware 2016 is between 2013 and 2017

**Comment by mbalous** [comment]
Windows Security just quarantied two of the plugins of ProcessHacker as _Trojan:Win32/Detplock._

- plugins\ExtendedNotifications.dll
- plugins\OnlineChecks.dll

![obrazek](https://user-images.githubusercontent.com/10358985/76142739-a2abe780-6070-11ea-9e41-4097ef61b845.png)

**Comment by dmex** [comment]
Isn't this the 9th "false positive" from Microsoft so far this year? and it's only March...

They did the same thing last Friday because they know nobody would be in the office over the weekend and it would prolong/maximize damage to the project. 

Also not surprised they would target the OnlineChecks plugin and automatically remove it from Windows (second time this week) when its required for integrating Process Hacker with 72 antivirus engines via the cloud....

...


![image](https://user-images.githubusercontent.com/1306177/76146959-8ac17b80-60e3-11ea-9098-c01a23bc9434.png)

**Comment by ygoe** [comment]
Looks like a targeted offence. Sounds to me like Microsoft is now officially evil. This is the kind of game that little children play. Windows Defender might no longer be a serious security solution but simply a tool to suppress undesired developer productivity apps. Have they learnt too much from Chinese politicians? Process Hacker should explain this offensive behaviour to their users prominently and strongly suggest not using Windows Defender if they seek for reliable and responsible protection.

**Comment by Laibalion** [comment]
And we're there again:
![image](https://user-images.githubusercontent.com/29300480/79192325-cb21c100-7e28-11ea-930e-8c41297ada9f.png)

This is for version 3.0.2993

**Comment by ExtremeGTX** [comment]
Just Downloaded the latest portable version from release page, and Windows defender didn't detect anything. 
Also after scanning the whole directory, everything is clean!.
Windows 10 1903 Build 18362.720

**Comment by poizan42** [comment]
This time it's the ExtendedTools plugin that's being detected:
![image](https://user-images.githubusercontent.com/308129/87541423-857fc900-c6a1-11ea-8b87-2b1dad0301a4.png)

**Comment by h-h-h-h** [comment]
Can somebody please explain why, on VirusTotal, the nightly downloads get assigned the tags `downloads-pe` and `downloads-zip`?

VirusTotal links:

- [Setup .exe (`downloads-pe`)](https://www.virustotal.com/gui/url/24594facebb8ab7705bc32059169ee093b6e799f3ab0fa4dea607530c649101f/detection)
- [Portable .zip (`downloads-zip`)](https://www.virustotal.com/gui/url/92e9a751c2bb5503f2dd429c8acb55c4bb97783ecbb9b198fc0adbd24227b3dc/detection)

**Comment by dmex** [comment]
> the nightly downloads get assigned the tags downloads-pe and downloads-zip

Those tags are assigned based on the HTTP headers returned by the URL rather than the actual file content because you're actually scanning the URL itself and not the downloaded file. If you want to scan the file downloaded by the URL you have to first download it then submit to VirusTotal rather than passing the URL.

**Comment by poizan42** [comment]
This is a whole new level of heuristics fail. A Linux backdoor in a Win32 DLL...
![image](https://user-images.githubusercontent.com/308129/92903453-65128780-f422-11ea-8464-77847fd4b879.png)

**Comment by Biswa96** [comment]
> A Linux/Android exploit in a Win32 DLL...

Well done! ProcessHacker is now officially a cross platform 100% free and open-source malware! 🥳

**Comment by dmex** [comment]
![image](https://user-images.githubusercontent.com/1306177/97745696-5ea8ae00-1b3d-11eb-8f41-49b9b0b8695b.png)

**Comment by garoto** [comment]
I guess at this point, this should be an issue for people who use PH on some sort of restrictive evironment, like a corporate network, because otherwise issues dealing with AV snake-oil vendors are just noise at this point.

**Comment by dmex** [comment]
The detection is still active: https://user-images.githubusercontent.com/1306177/98056549-45cd2f00-1e94-11eb-83c9-402cd9582e09.png

> some sort of restrictive environment, like a corporate network, 

Defender is built-in and technically cannot be disabled so that's an issue for everyone. How are you going to use Process Hacker when Microsoft keeps deleting the binaries claiming they're malware?

**Comment by ralvarador** [comment]
I worked in two companies that its AV were set to block Process Hacker: Kaspersky and Sophos...

**Comment by ygoe** [comment]
Wow, that AV thing doesn't seem to be about "viruses" and malware anymore but rather opinionated. Like so many sad things these days.

This is my standard Process Hacker installation procedure:

1. Install Process Hacker
2. Add the installation directory to Defender's exclusion list (like Thunderbird's mail storage to prevent total mail corruption, theoretically)
3. Install Process Hacker

Maybe the website and/or setup package could promote or even automate that, too.

**Comment by Shirk** [comment]
I'd be okay if it was a Windows Defender only false positive.. but this is a joke:

![image](https://user-images.githubusercontent.com/304352/98271236-a1c4b000-1f87-11eb-9f62-7cc23eada8c5.png)

**Comment by luckydonald** [comment]
"Riskware.Processhacker" Wow.

**Comment by Dwarden** [comment]
Windows Defender reports those
processhacker-3.0.3585-setup.exe  = Trojan:Win32/CryptInject!ml
ProcessHacker.exe , ProcessHacker.lnk = Trojan:Win32/Zpevdo.B
processhacker-3.0.3585-bin.zip = Trojan:Win32/Detplock
replace .zip files with .7z
please sign installer and processs binary with same certificate as the driver (or w/e trustable)

**Comment by Dwarden** [comment]
Windows Defender changed the detection and reports those
processhacker-3.0.3593-setup.exe = Trojan:Win32/Woreflint.A!cl
processhacker-3.0.3593-bin.zip = Trojan:Win32/CryptInject!ml

**Comment by dmex** [comment]
@luckydonald 

Wen Jia (wj32) was an employee for Webroot and worked in their US offices for a few years - they issued the "Riskware.Processhacker" signature the same day he left the company.

They're exploiting the legal immunity given by Section 230 in the USA to attack former employees and others without justification.

@Dwarden 

> please sign installer and processs binary with same certificate as the driver 

The binaries are already signed. The problem is Microsoft has added signature checks to a dozen Windows API functions that are explicitly targeting/blocking Process Hacker (separate from Defender) and have refused to fix other highly critical Windows issues.... Signing won't fix anything until Microsoft changes its illegal policies.

The browser wars have ended but the task manager wars are still going strong... taskmgr.exe has been hard-coded throughout multiple components to prevent users from changing the system defaults to third party programs just like Microsoft did with Internet Explorer - the signature checks included with API functions used by taskmgr.exe designed to block competitors is much worse than attacks by Microsoft on Netscape in the 90's and other APIs have been changed to report incorrect information when used by third parties but not their own products.

The malware detections by Microsoft are part of this and are being used as a pre-emptive attack aimed at diminishing our legitimate claims against the company for anti-trust violations.

2008-2018  - 0 false positives by Defender over those 10 years.
2019 - 10 or so false positives
2020 - well over 35 'false positives'.

The first "false positive" by defender was just a few days after wj32 quit working at Microsoft. Mark Russinovich sent both of us job offers but I turned it down because of work on critical issues at Electronic Arts. David Hepkin from the Azure/hyperv/vsm team was wj32's team leader.

Microsoft targeting former employees is one thing but targeting competitors products is well established as anti-competitive by the courts and regulators so the defender team are including our binary signatures in random detections in an attempt to conceal the purpose of the attacks and there are also potentially cases with employees "stat padding" to obtain contract KPI bonuses.

"stat padding" is a reason behind random defender signatures (e.g. Linux/Lotoor.A) containing signatures for Process Hacker despite there being zero relationship - security vendors have contract bonuses for the number of detections and employees 
 are using competitors binaries to increase the # of detections and obtain those bonuses. This is also why the same signature was used against 6 separate open source projects over 6 months without any shared source code which is statistically impossible for an accidental 'false positive'.

Microsoft still hasn't provided the team with any evidence to support their claims... The best we've received is a regurgitated report from another Microsoft employee here in Melbourne, Australia which made no sense.

MSRC also hasn't responded to emails about these and other issues... These attacks by Microsoft are likely going to get worse until the courts/regulators are more involved so I recommend users switch to Malwarebytes: https://www.malwarebytes.com/

Malwarebytes has been supportive of Process Hacker and they resolve issues extremely quickly in less than two hours - a stark contrast to the actions by Microsoft.

**Comment by namazso** [comment]
@dmex 

About those "!cl" and "!ml" ending detections on Defender, my softwares get that all the time too. They're probably machine learning or some other dumb heuristic. For me it always resolves after submitting, but it's a bit tiresome to do that every single version. (I'd also recommend only caring about it when it's release time, not for random builds, that's just not worth the time.)

However I'm not entirely sure about that Malwarebytes recommendation. They have no way to submit false positives without registering to their forum and stuff (eg. via a form or email). Clearly, they don't really care about software developers trying to get their one of a dozen AV's detection cleared, only about those who use (and often pay) for their software.

Also, do you have any source for the taskmgr signature checks? (source as in what binary has this? I have legitimate interest, not just accusing you or something)

**Comment by Dwarden** [comment]
unfortunately there are some factual errors on your side
inbetween the binaries delivered with nightly installer and .zip file 
processhacker-<%buildnumber%>-setup. itself IS NOT signed
processhacker.exe IS NOT signed
peid.exe IS NOT signed
any <%plugins%>.dll ARE NOT signed
the ONLY signed file IS the `kprocesshacker.sys`
with certificate from 2016 including the long deprecated SHA1 and SHA256 (dual sign certificate)
(some AV companies seems to auto-flag binaries with dual sha-1 and sha-256 cert since last year)
> certificates and intermediates signed in SHA1 won't be recognized anymore and will provoke security alerts
> https://docs.microsoft.com/en-us/sysinternals/announce/sha1deprecation

my suggestion
phase out the dual signature and replace it with fresh new certificate, using just SHA256
then sign all .exe , .dll , .sys files , no exceptions nor leftovers, with this new certificate
note: legacy OSses like XP SP3, Vista, Windows 7 and Server 2008 support SHA256 with correct KB installed

after this is done we can continue argument about how AV/security or Microsoft handles detection
if this isn't done all they can do, is point to what i just posted as excuse argument

**Comment by Dwarden** [comment]
Windows Defender changed the detection and reports those
processhacker-3.0.3612-setup.exe = Trojan:Win32/Spursint.F!cl
processhacker-3.0.3612-bin.zip = Trojan:Win32/Spursint.F!cl
32bit ProcessHacker.exe = Trojan:Win32/CryptInject!ml

**Comment by VictorVG** [comment]
People "earn honestly" for bread and butter and girls, and therefore do not be surprised - they will find something else.:)

**Comment by dmex** [comment]
@Dwarden 

> Windows Defender changed the detection and reports those

I need screenshots of these detections so they can be used as evidence. WinDefThreatsView from Nirsoft shows all the relevant detection information that we need: http://www.nirsoft.net/utils/windows_defender_threats_view.html


> unfortunately there are some factual errors on your side
> processhacker-<%buildnumber%>-setup.exe IS NOT signed
> processhacker.exe IS NOT signed


Thanks captain obvious. 

If you haven't been paying attention... Microsoft changed the signing requirements for Windows back in 2015 requiring developers to submit binaries to Microsoft and receive 'attestation signed' copies of those binaries:
https://docs.microsoft.com/en-us/windows-hardware/drivers/dashboard/attestation-signing-a-kernel-driver-for-public-release

Attestation signing is another policy Microsoft uses against competitors in the Task Manager wars... To be specific:

**Microsoft does not sign our binaries with the correct certificate attributes in order to block and/or severely limit competitors from calling most API functions used by Task Manager and Process Explorer.**

The same is true for the following products and not only Process Hacker:  (no specific order)
* AnVir Task Manager
* MiTeC Task Manager
* System Explorer
* Daphne
* WinUtilities Process Security
* Process Lasso
* TaskInfo
* DTaskManager
* Auslogics Task Manager
* Also every other competitor.

All of these products are affected and also incorrectly signed blocking them from using dozens of new Windows features and API functions (ntdll/user32) by Microsoft that are being used by Task Manager and Process Explorer.

I'm not sure why those other competitors support that Microsoft policy or why they've been silent about it... I personally refuse to pay $1500 a year for an illegal anticompetitive certificate until it includes the attributes required for features used by Task Manager and Process Explorer.

Even If I wanted to submit the binaries; the submission fails with an unknown error and Microsoft have never corrected either issues with two seperate supports going nowhere. That is effectively banning my account from ever signing a Process Hacker release. 

The signing submissions are still in limbo even now:
![image](https://user-images.githubusercontent.com/1306177/101326670-b7e4c980-38c1-11eb-8245-babeb32d4b08.png)

David Plummer, the former Microsoft developer who wrote Windows Task Manager has also been made statements in the media earlier this year about "secret code" he embedded into Windows giving Microsoft an unfair competitive advantage... I've been discussing Microsoft's actions and policies with our competition regulator here in Australia but its a slow process and I'll have more to share soon™

**Comment by Dwarden** [comment]
@dmex primary i wasn't writing about driver but binaries and libraries in the nightly

i.e. you mention `Process Lasso`  yet that one has ALL binaries and libraries signed for years
Process Lasso works fine, no WindowsDefender detection on any installer, download or install itself nor execution of binaries ...
several of the other software from the list aren't detected by Windows Defender either but I've got no time to test each
e.g. fire are TaskInfo, Daphne etc.

anyway because you decided to aggressively respond to simple request and comments mainly about unsigned binaries
i will not continue in this discussion further

**Comment by dmex** [comment]
> anyway because you decided to aggressively respond

I'm Australian 🤷‍♂️

It wasn't an intention to be aggressive but these are serious issues. Microsoft and other vendors requested Github delete the project for Terms of Service violation earlier this year and there was an attempt to seize our website domain so I've decided to speak out more about the issues we deal with managing this project and the problems we're having going forward and the things we're told by people at those companies.

The webpages published by vendors (i.e. ProcHack!MTB) are used as reference material by other organisations as is this thread on the tracker discussing the issue and the comments weren't aimed at you and I apologise for not making that more clear.

> you mention Process Lasso yet that one has ALL binaries and libraries signed for years

The Process Lasso executable certificate hasn't been signed with the correct attributes for over a decade. Microsoft hasn't documented these restrictions so they either don't know or don't care.

They're blocked from calling Windows APIs used by Task Manager like CreateWindowInBand, classes found in NtQuerySystemInformation/NtSetSystemInformation and from using the "secret code" mentioned by Plummer in csrss/winlogon.

If they're serious about competing with Task Manager and Process Explorer then these are some of the Windows features they would probably want to include but can't because of Microsoft's anti-competitive policies.

> several of the other software from the list aren't detected by Windows Defender either 

Microsoft added a new detection "`ProcHack!MTB can be used to patch or "crack" some software so it will run without a valid license or genuine product key.`" https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?Name=HackTool:Win32/ProcHack!MTB&ThreatID=2147755297

The development team does not and has never supported piracy. There has never been anything like that ever included or ever supported by this project. No such feature or functionality for patching or cracking software has ever existed in Process Hacker.

**Comment by dmex** [comment]
> anyway because you decided to aggressively respond

I'm Australian 🤷‍♂️

It wasn't an intention to be aggressive but these are serious issues. Microsoft and other vendors requested Github delete the project for Terms of Service violation earlier this year and there was an attempt to seize our website domain so I've decided to speak out more about the issues we deal with managing this project and the problems we're having going forward and the things we're told by people at those companies.

The webpages published by vendors (i.e. ProcHack!MTB) are used as reference material by other organisations as is this thread on the tracker discussing the issue and the comments weren't aimed at you and I apologise for not making that more clear.

> you mention Process Lasso yet that one has ALL binaries and libraries signed for years

The Process Lasso executable certificate hasn't been signed with the correct attributes for over a decade. Microsoft hasn't documented these restrictions so they either don't know or don't care.

They're blocked from calling Windows APIs used by Task Manager like CreateWindowInBand, classes found in NtQuerySystemInformation/NtSetSystemInformation and from using the "secret code" mentioned by Plummer in csrss/winlogon.

If they're serious about competing with Task Manager and Process Explorer then these are some of the Windows features they would probably want to include but can't because of Microsoft's anti-competitive policies.

> several of the other software from the list aren't detected by Windows Defender either 

Microsoft added a new detection "`ProcHack!MTB can be used to patch or "crack" some software so it will run without a valid license or genuine product key.`" https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?Name=HackTool:Win32/ProcHack!MTB&ThreatID=2147755297

The development team does not and has never supported piracy. No such feature or functionality for cracking Microsoft software or activations has ever existed in Process Hacker.

**Comment by Marvin-Brouwer** [comment]
I also get a report on this using Sophos:
https://search.sophos.com/#q=processhacker&t=Support&sort=relevancy

**Comment by dmex** [comment]
> I also get a report on this using Sophos

Sophos are irrelevant considering they're also targeting and blocking Microsoft software such as Process Explorer:

https://www.sophos.com/en-us/threat-center/threat-analyses/controlled-applications/Process%20Explorer.aspx
https://www.sophos.com/en-us/threat-center/threat-analyses/controlled-applications/Sysinternals%20Desktop.aspx
https://www.sophos.com/en-us/threat-center/threat-analyses/adware-and-puas/PsExec.aspx

**Comment by VictorVG** [comment]
This is how Sophos solves its private commercial problem - to increase sales of its software.

In 1993, I had a conversation with Eugene Kaspersky on a similar occasion - then there was a "antivirus" AidsTest by Mr. Lozinsky that was updated as soon as a new virus appeared. And then Eugene said - “It's strange, it takes several days to analyze a new virus and find a way to eliminate it, and a new version of AidsTest comes out a few minutes after the appearance of a new infection, and a significant part of its database is the first programs that came across. the remedies for his treatment are written by the same person to increase the sales of his program, and for the same purpose, random harmless programs are added to the AidsTest database ... ". And after that, similar questions arose about a number of other antiviruses, but there was no direct connection "virus - antivirus creation", although there were plenty of facts of adding random harmless programs to the AV database.

Now other antivirus manufacturers use a similar method to increase their sales - they add randomly selected popular programs to their AV databases, and they also do not hesitate to publish deliberately false messages about the detection of "viruses" in the media. Naturally, anonymous "experts of antivirus companies" or "experienced users" who check numerous user complaints about the malware's harmfulness become the source of information.

So recently a campaign was organized to discredit WinRAR v5.91, and now it is being repeated in relation to v6.00 - there are already messages on forum.ru-board.com about the deletion of RAR 6.00 archives with text files under the pretext "a virus, trojan, malware "a number of antiviruses.

The same strategy is used with respect to RN - why invent something new if the statistics of the number of installations of their product is increasing, and it does not matter that the authors of the declared malicious programs are forced to spend time and money on proving the absurdity of the accusations made on behalf of anonymous users and reworking their software in order to Eliminating false positives from antivirus software is just business and nothing personal.

And the reasons for this behavior back in the 18th century were explained by the British publicist Thomas Joseph Dunning:

> "Capital," says the Quarterly Review, "avoids noise and abuse and has a fearful nature." It's true, but it's not the whole truth. Capital is afraid of lack of profit or too little profit, as nature is afraid of emptiness. But once there is sufficient profit, capital becomes bold. Provide 10 percent, and capital agrees to any use, at 20 percent it becomes animated, at 50 percent it is positively ready to break its head, at 100 percent it tramples on all human laws, at 300 percent there is no crime that it would not risk, even on pain of the gallows. Proof: smuggling and slave trade.

**Comment by dmex** [comment]
Process Hacker was targeted by SolarWinds/Sunburst/Teardrop: 
2032008861530788751:processhacker

Russian anti-virus company Doctor Web has today classified Process Hacker as malicious malware: 
Tool.ProcessHacker.2(7), 
Tool.ProcessHacker.3, 
Tool.ProcessHacker.4(2),
Tool.ProcessHacker.5

Attacked by vendors on one side and attacked by malicious software on the other side... plus covid-19 and lockdowns... 2020 has been the absolute worst 😕

**Comment by ygoe** [comment]
Just an idea: would it be possible to produce a "light" version of Process Hacker that has the same UI and similar feature set, just without the problematic parts that allow these false classifications? I use PH as everyday process management tool, running as admin, but not going too deep into the guts of the processes. What are the special features of PH that require those internal APIs or knowledge of Windows, or the signed driver?

**Comment by VictorVG** [comment]
This proposal will not work because the "hunt" is not for a mythical virus, but for the project itself as a competitor to paid solutions of monopolies, and it is they who are the real initiators and profitable events. Therefore, you can cut out the possibilities as much as you like, but they will not be satisfied with this until they destroy any idea of ​​creating alternatives to their paid products. It's just a business, nothing more. And Mr. Mostovoy's DrWeb was distinguished by attentiveness to his "wishes" back in the early 90s, when its antivirus was better known in Russia than in the world.

**Comment by dmex** [comment]
> would it be possible to produce a "light" version of Process Hacker that has the same UI and similar feature set, just without the problematic parts that allow these false classifications? 

We're planning on a few things to remedy the situation but removing functionality is not one of them.

> What are the special features of PH that require those internal APIs or knowledge of Windows, or the signed driver?

It is impossible to compete with Task Manager and Process Explorer without using internal APIs... Microsoft does not allow anyone to compete with those products and literally everything used by Task Manager and Process Explorer is undocumented.

Yeah sure you can use EnumProcesses from psapi and some others that are documented but they're not used by Task Manager and Process Explorer and it does not return anywhere near the same information as those products. PSAPI also suffers from compatibility issues (those functions were designed for Windows 95!) and they're completely unusable for security checks required on newer versions of Windows.

> you can cut out the possibilities as much as you like, but they will not be satisfied with this until they destroy any idea

This is basically what's happened with KPH... It's just a tiny shell of what it used to be and they keep making more demands. 

> Mr. Mostovoy's DrWeb was distinguished by attentiveness to his "wishes" back in the early 90s, when its antivirus was better known in Russia than in the world.

I found the timing particularly interesting. Process Hacker was targeted by SolarWinds/Sunburst/Teardrop - DrWeb haven't released signatures for SolarWinds/Sunburst but decided to release signatures for Process Hacker? 🤔

![image](https://user-images.githubusercontent.com/1306177/103102601-eb20aa00-4670-11eb-8ed1-ce929fee1b1b.png)

**Comment by VictorVG** [comment]
In the early 90s, DrWeb was the "main alarmist" on the Nowell NetWare platform, and now there are many complaints about it from different users. The main problem is to get an intelligible answer from them to a clearly asked question.:)

**Comment by dmex** [comment]
* Microsoft has released two new signatures today:
Trojan:Win32/Wacatac.B!ml - peview  - https://i.imgur.com/TdsIGRV.png
Trojan:Win32/Warflint.A!cl - installer - https://i.imgur.com/hD0liZD.png

* ExtendedTools has been classified malicious:
Win64:MalwareX-gen 
Trojan.Win32.Johnnie
 https://i.imgur.com/m5Yb3Nh.png

* CDRF has declared our website malicious and have also refused our request to review the classification:

![image](https://user-images.githubusercontent.com/1306177/103259776-ed776100-49ee-11eb-8f8c-4312509b4866.png)

**Comment by dmex** [comment]
CDRF have now suddenly removed the classification for our website 🤔 🤷‍♂️

![image](https://user-images.githubusercontent.com/1306177/103261567-1bac6f00-49f6-11eb-83f9-f8c0539999a3.png)

**Comment by ralvarador** [comment]
Today... ehem... yesterday... anyway, I received this message when updating...
The update was sent to quarantine immediately! 
![imagen](https://user-images.githubusercontent.com/30580377/103263644-769e8180-4976-11eb-88a6-97a78e5eb98e.png)

**Comment by VictorVG** [comment]
This is how they react to the name "Process Hacker". For the sake of interest, I took the code from S/360 and entered into it "Process Hacker" - 13/70 Microsoft, Sophos and K° - "virus, trojan, keylogger, dialer, password sniffer".:)

**Comment by enervated** [comment]
I don't see a way to attach images here so I guess I will link to the screenshot dmex asked me to share a couple days ago:

https://wj32.org/processhacker/forums/download/file.php?id=1101&sid=9614de581af56890dfad9239c066339d&mode=view

From this thread:

https://wj32.org/processhacker/forums/viewtopic.php?f=40&t=4342

**Comment by Chaython** [comment]
Windows and many virus scanners hate that the application is unsigned. If the application was signed, it would likely be trusted.
Some stupid scanners hate the word "hacker".

**Comment by VictorVG** [comment]
But this - the word "hacker" in the name can become a trigger for other flywashers - "No need to check, we kill by name!" Once upon a time, Symanteck by the name .ko in make "found" viruses in the FreeBSD UNIX source and ... erased them along with local ports, and after two years the developers of the fly swatter convinced me that "Once removed, it means a virus! antivirus is not wrong!".:)

**Comment by enervated** [comment]
Nightly build 3.0.3701 (more specifically, ProcessHacker.exe) is detected as "Trojan:Win32/CryptInject!ml" by Windows Defender just like several previous versions.

**Comment by trueTrueman** [comment]
Kaspersky Internet Security with today's updates just started reporting 2.39 main exe and setup//data0006 as Trojan.Win32.Shelma.bcti

**Comment by tiziodcaio** [comment]
Trojan:Win32/Wacatac.B!ml detected in nightly version 3.0.3720 today.

**Comment by Ciantic** [comment]
I guess there is not much one can do than trust this.

To me file nightly gives this: Win32/Woreflint.A!cl
SHA256 of the portable zip 6104DCA0AF58911A9D0835C15B849754BBBE23F1C9EAF01C7EA41D50AFD007A1

But I suppose it too is an false positive.

**Comment by dmex** [comment]
The zip file currently has 36 detections that are all completely different... I haven't been able to get a sensible response from anyone about it and nobody seems to know what's happening.

Half of the detections are:
Win32:MalwareX
Win32.Troj.Generic_a.a
Artemis!81B884FCC970
PUP/Win32.RL_Generic.R359108
Gen:Variant.Mikey.118476
Riskware/ProcHack
not-a-virus:HEUR:RiskTool.Win32.Proc
Malware.AI.1270165517

It's all over the place with every vendor detecting something completely different:
https://www.virustotal.com/gui/file/6104dca0af58911a9d0835c15b849754bbbe23f1c9eaf01c7ea41d50afd007a1/detection

The setup - which has an embedded copy of the _exact same zip file_ - has only 9 detections but they're all different yet again:
https://www.virustotal.com/gui/file/d2dcdca30553dc0a99e541fb19d4af6735ac165be8a0becbf20f9a1df023372b/detection

AV software is definitely broken right now... I've discovered the "Gen:Variant.Mikey" detections are because we're using the [MessageBoxW function](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-messageboxw)... the others I have no idea.


An educated guess is that vendors updated their sandboxes/scanners with the latest Windows patches released the other day but then haven't bothered to account for internal API changes introduced by Microsoft and are now detecting those internal changes as modifications by the process being scanned 🤷‍♂️

I'm going to replace MessageBoxW as a workaround for the "Gen:Variant.Mikey" detections but that might cause issues with the warning dialogs/error prompts... I'll try get some sense from vendors why their software is broken and when they're releasing fixes.

**Comment by dmex** [comment]
Microsoft and Malwarebytes have removed their detections for the latest nightly... I've also removed MessageBoxW and MessageBeep from the IAT.

That's 5 detections removed right there, 30 more vendors making excuses and arguments yet to go.

What an absolute fucking joke of an industry.

**Comment by artem-kamyshev** [comment]
Hello,

@dmex Today Kaspersky Internet Security has "cured" (that is removed) ProcessHacker on my Windows 10 PC. It's component "Applications' Control" has detected via Cloud Protection that ProcessHacker is an "VHO:Trojan.Win32.Agent.xagpxa".

I've reinstalled it without running right after installation and added ProcessHacker to 'Trusted Zone'. That let me run ProcessHacker again. At least for some time as its whole directory was already in an exclusion list of a File Antivirus component and that was enough until this day.

Hope this info helps somebody who would face the same problem.

I've attached a screenshot and a report in case you'll go discuss the issue with Kaspersky. They are in Russian though.

[kav-1.txt](https://github.com/processhacker/processhacker/files/5979913/kav-1.txt)

![изображение](https://user-images.githubusercontent.com/31185325/107914608-d9be9480-6f73-11eb-8fa4-e325cc46bafa.png)

**Comment by yfdyh000** [comment]
> Hello,
> 
> @dmex Today Kaspersky Internet Security has "cured" (that is removed) ProcessHacker on my Windows 10 PC. It's component "Applications' Control" has detected via Cloud Protection that ProcessHacker is an "VHO:Trojan.Win32.Agent.xagpxa".
> 
> I've reinstalled it without running right after installation and added ProcessHacker to 'Trusted Zone'. That let me run ProcessHacker again. At least for some time as its whole directory was already in an exclusion list of a File Antivirus component and that was enough until this day.
> 
> Hope this info helps somebody who would face the same problem.
> 
> I've attached a screenshot and a report in case you'll go discuss the issue with Kaspersky. They are in Russian though.
> 
> [kav-1.txt](https://github.com/processhacker/processhacker/files/5979913/kav-1.txt)
> 
> ![изображение](https://user-images.githubusercontent.com/31185325/107914608-d9be9480-6f73-11eb-8fa4-e325cc46bafa.png)

+1. The https://processhacker.sourceforge.io/nightly.php looks fine.

**Comment by garoto** [comment]
Welp, at the very least, you now achieved "closure" in regards how these scammy "tech" conglomerates sees your software and you can now move forwards, and hopefully tagging any issues dealing with AV false-positives as "invalid".

**Comment by gchq-zz** [comment]
Since I whitelisted it there has not a been a problem, it's run flawlessly since this debacle started

**Comment by asheroto** [comment]
Malwarebytes and Webroot report this too, and apparently more...

VirusTotal report:
https://www.virustotal.com/gui/file/70211a3f90376bbc61f49c22a63075d1d4ddd53f0aefa976216c46e6ba39a9f4

**Comment by VictorVG** [comment]
What about thinking? Someone needs competitors with much more capabilities than their programs?

**Comment by asheroto** [comment]
I would like to understand, but don't understand what you mean. 🙂

**Comment by Masamune3210** [comment]
There isnt any malware, antimalware companies are just dumb and Microsoft is anti-competetive

**Comment by asheroto** [comment]
Something else that might help, at least for the official release, is for the developer to use a current valid code signing certificate. The current one on kprocesshacker.sys expired in 2017...

![image](https://user-images.githubusercontent.com/49938263/162236284-2978362b-371f-4a88-9556-6ec76b40450c.png)

I'm totally not advertising, my company has a partnership with DigiCert so I can have them issue an EV certificate for $485/year as opposed to the retail cost of $664/year for an EV cert. That's over $500 off for a 3 year, ha! You can find me on [asheroto.com](https://asheroto.com).

**Comment by ygoe** [comment]
@asheroto The certificate issue has been discussed already, maybe in this issue. And it's not "expired", the signature was created within the time of validity of the certificate so it's alright. There just hasn't been an update to it anymore, but that's not necessary for signatures.

**Comment by asheroto** [comment]
Very true, I suppose I was wondering if it would help curb some of the detections but no way to really know.

**Comment by Masamune3210** [comment]
Best way for the common user to help is report the false identification to virus databases, since their dumb heuristic engines can't seem to understand that strong privileges does not mandate something to be a virus

**Comment by VictorVG** [comment]
Process Hacker uncovered gross errors in the Valve Corporation software, which became the reason for its war against the project, which was later joined by antivirus programs, as well as Microsoft Corporation, which also has its own interests in it - people pay for a license for its Task Manager when buying an OS as well , as previously paid for Internet Explorer - according to Microsoft, these programs are not free.

**Comment by gchq-zz** [comment]
Back to my comment just over a year ago

'Since I whitelisted it, there has not a been a problem, it's run flawlessly since this debacle started'

Still working fine on Win 10 without any issues

**Comment by sabihoshi** [comment]
Just so you guys know, every time there is a comment about 42 people gets emailed about this issue, maybe this should be locked as this problem is resolved (that is, being reported and removed by Windows Defender)
![image](https://user-images.githubusercontent.com/25006819/162576353-f655d456-9730-4636-b192-6f028db35d4e.png)

**Comment by krokofant** [comment]
@sabihoshi People can edit their notification settings.
![image](https://user-images.githubusercontent.com/5908498/162584256-2bf377f2-cf67-4dfe-a85f-dad52b4d9ee8.png)

No need to block any further comments just because some people get notifications. Those that actively follow this issue might realize this likely won't be fixed or closed, and if you realize that then you might as well unsubscribe from the issue if you don't want to take part in it.

**Comment by Black-Tide** [comment]
> Just so you guys know, every time there is a comment about 42 people gets emailed about this issue, maybe this should be locked as this problem is resolved (that is, being reported and removed by Windows Defender) ![image](https://user-images.githubusercontent.com/25006819/162576420-87dc72b1-9d22-4e42-ae94-4db759b5aec4.png)

No its not. I just had Windows Defender go nuts for the same made up virus in the nightly build of Process Hacker 3 that I've been using for a week and Defender was chill with until it forgot to take it's Bipolar meds: 

Trojan:Win32/Wacatac.B!ml

They said they were fixing this 3 years ago and yet they link me to the same old article from 2019.

**Comment by vharmash** [comment]
Hi folks, looks like the Defender strikes back again:
![image](https://user-images.githubusercontent.com/69356233/220929818-ce308eab-a53b-4b3f-87e3-57f89e876627.png)

**Comment by VictorVG** [comment]
And think? Little the inc? Or as one "specialist" writes:

"I am a programmer of the old school. I don’t read anything. I only write. If crap is not clear how to use the poke method, then well, this crap, this crap."

original, in Russian - "Я -- программист старой школы. Я ничего не читаю. Я только пишу. Если хренью не понятно как пользоваться методом тыка, то ну ее нах, эту хрень. "

:lol:

Did you decide that the first one encountered the "unknown"?

**Comment by dmex** [comment]
@vharmash 

Report false positives to Microsoft by uploading the binary here:
https://www.microsoft.com/en-us/wdsi/filesubmission

The same way you report binaries is the exact same way we report binaries 🤷‍♂️

**Comment by asheroto** [comment]
Gotcha, thank you.

**Comment by ch3rry8om8** [comment]
my win 10 allowed system informer to run but as soon as i opted to make it my default task manager  defender quarantined it as Trojan:Win32/Bearfoos.A!ml
